src/lib/db.ts

```diff
@@ -1,101 +1,133 @@
 import Dexie from 'dexie';
-import _ from 'lodash';
-import { canonicalizeUrl } from './url.js';
+import { concat, map, sumBy } from 'lodash';
+import { canonicalizeUrl } from './url.ts';
+import { isBackgroundPage, isTesting } from './util.ts';
+import {
+  IActivity,
+  IDonation,
+  Donation,
+  IThank,
+  Thank,
+  ICreator,
+  Creator,
+} from './models.ts';
 import isReserved from 'github-reserved-names';
-import { Donation, Thank } from './models.js';
 
-let _db = undefined;
+// A decorator that will throw an error if the decorated
+// function is called outside of the background script.
+function onlyInBackgroundPage(): any {
```

Comment:
I thought this was super clever. Great way to prevent programmer errors.

Next challenge: Make it automatically dispatch the request to the background page if not on the background page.